### ISSUE TITLE
Block crashes while editing if inside <a>element is a placeholder without attributes

### ISSUE THREAD

**Issue by dozentis** [WordPress/gutenberg#38815]
### Description
A text was copied/pasted from a Word document (with headings) into the Block-Editor. Therefore, an anchor with name attribute is visible in the HTML of the heading, because the headings are linked to the table of content. 
After inserting the copied MS-Word text into the block editor, Gutenberg removes the name-attribute and leaves a blank <a> element inside the <h> element. This causes a crash of the heading-block when editing the linked heading-text. (crash only reproducible in WP 5.9)

### Paste MS Word sample-text into html-editor, result:

`<h1><a name="_Toc95814112"></a>Heading 1</h1>`
### Paste sample-text into block-editor, result :
Gutenberg removes name-attr
````
<!-- wp:heading {"level":1} -->
<h1 id="heading-1"><a>Heading 1</a></h1>
<!-- /wp:heading -->
````

## STR
1. Download attached _sampletext-with-toc.docx_
2. Open with MS Word (not google-doc)
3. Mark and copy the text after generated "Table of Content"
4. (not a necessary step, just for checking) Paste copied text into your html-editor and notice that there are empty anchors in <h>-elements with name-attributes included by MS Word
5. Open new WordPress Page or Post (Only WordPress V5.9!)
6. Just paste the copied text.
7. Notice that the heading will have (an empty) link by their hover-color
8. Click into the heading-text for editing (not after)
9. Heading Block crashes

### Additional info WP 5.8: 
Until WP 5.8 i've often used this method to copy/paste from a Word-document and it worked! Gutenberg removed all unneeded stuff.  I only needed to click (after pasting) into the heading and removed the empty link.

### Screenshots, screen recording, code snippet

[sampletext-with-toc.docx](https://github.com/WordPress/gutenberg/files/8068015/sampletext-with-toc.docx)

### Environment info

WP 5.9 / Twenty Twenty One and other themes
Windows 10 / Chrome newest

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by dozentis** [comment]
Easier testing, just copy/paste this code

```html
<!-- wp:heading {"level":1} -->
<h1 id="heading-1"><a>Heading 1</a></h1>
<!-- /wp:heading -->
```

into an empty line in block editor (not code-editor) and click into the headline text

**Comment by dozentis** [comment]
console log
`TypeError: t.url is undefined in react-dom.min.js:141:274`

```
react-dom.min.js?ver=17.0.1:141 TypeError: Cannot read properties of undefined (reading 'length')
    at Dn (block-editor.min.js?ver=56fd274d0d14fe0540d61051a6a899d2:2:103142)
    at Le (react-dom.min.js?ver=17.0.1:100:3)
    at Pj (react-dom.min.js?ver=17.0.1:231:183)
    at di (react-dom.min.js?ver=17.0.1:168:305)
    at Nj (react-dom.min.js?ver=17.0.1:168:236)
    at sc (react-dom.min.js?ver=17.0.1:168:96)
    at gf (react-dom.min.js?ver=17.0.1:162:109)
    at react-dom.min.js?ver=17.0.1:73:230
    at unstable_runWithPriority (react.min.js?ver=17.0.1:24:26)
    at Za (react-dom.min.js?ver=17.0.1:73:8)
```

**Comment by dozentis** [comment]
## addendum: Same issue with paragraph-block

````
<!-- wp:paragraph -->
<p><a>I will crash the block</a></p>
<!-- /wp:paragraph -->
````

I changed the title of this issue

**Comment by mcsf** [comment]
Hi, @dozentis. Thanks for your report.

This issue was recently fixed in #37034. Right now, the bug affects WP 5.9, but the fix should appear in the upcoming WP 5.9.1 version, as well as in the next stable Gutenberg plugin release.

_Closing: fixed by #37034._
